src/libAtomVM/module.h

```diff
@@ -262,6 +282,89 @@ static inline const uint8_t *module_get_str(Module *mod, size_t offset, size_t *
+    int best_label = -1;
+    const uint8_t *table_data = (const uint8_t *) this_module->export_table;
+    int exports_count = READ_32_ALIGNED(table_data + 8);
+    for (int export_index = exports_count - 1; export_index >= 0; export_index--) {
+        int fun_atom_index = READ_32_ALIGNED(table_data + (export_index * 12) + 12);
+        int fun_arity = READ_32_ALIGNED(table_data + (export_index * 12) + 4 + 12);
+        int fun_label = READ_32_ALIGNED(table_data + (export_index * 12) + 8 + 12);
+        if (fun_label <= label && best_label < fun_label) {
+            best_label = fun_label;
+            *arity = fun_arity;
+            *function_name = module_get_atom_string_by_id(this_module, fun_atom_index);
+        }
+    }
+
+    table_data = (const uint8_t *) this_module->local_table;
+    int locals_count = READ_32_ALIGNED(table_data + 8);
+    for (int local_index = locals_count - 1; local_index >= 0; local_index--) {
+        int fun_atom_index = READ_32_ALIGNED(table_data + (local_index * 12) + 12);
+        int fun_arity = READ_32_ALIGNED(table_data + (local_index * 12) + 4 + 12);
+        int fun_label = READ_32_ALIGNED(table_data + (local_index * 12) + 8 + 12);
+        if (fun_label <= label && best_label < fun_label) {
+            best_label = fun_label;
+            *arity = fun_arity;
+            *function_name = module_get_atom_string_by_id(this_module, fun_atom_index);
+        }
+    }
+    if (UNLIKELY(best_label == -1)) {
+        // Couldn't find the function.
+        return false;
+    }
+    return true;
+}
+
+/*
+ * @brief Insert the instruction offset for a given module at a line reference instruction.
+ *
+ * @details This function is used when loading a module.  When a line instruction is
+ * scanned, this function is used to record the instruction offset at which the line
+ * instruction occurred.
+ *
+ * Note that if the module (BEAM file) does not contain a Line chunk, then this function
+ * is a no-op.
+ *
+ * @param mod the module
+ * @param line_ref the line reference (index)
+ * @param offset the instruction offset at which the line instruction occurred.
+ */
+void module_insert_line_ref_offset(Module *mod, int line_ref, int offset);
+
+/*
+ * @brief Find the latest line reference (index) before or at which the instruction offset
+ * occurs.
+ *
+ * @details This function searches for the most recent line instruction scanned during
+ * code loading at or before the specified instruction offset.  This function is used to
+ * locate the line number from a continuation pointer in a stack trace.
+ *
+ * @param mod the module
+ * @param offset
+ * @return the line reference
+ */
+int module_find_line(Module *mod, int offset);
+
+/**
+ * @return true if the module has line information, false, otherwise.
+ */
+static inline bool module_has_line_chunk(Module *mod)
+{
+    return !IS_NULL_PTR(mod->line_refs);
```

Comment:
IS_NULL_PTR macro makes use of LIKELY, so I wouldn't use this, but I'd rather use `mod->line_refs != NULL`

Reply:
done